test/integration/remote-config.spec.ts

```diff
@@ -143,6 +157,87 @@ describe('admin.remoteConfig', () => {
           expect(template.conditions).to.deep.equal(VALID_CONDITIONS);
           expect(template.parameters).to.deep.equal(VALID_PARAMETERS);
           expect(template.parameterGroups).to.deep.equal(VALID_PARAMETER_GROUPS);
+          expect(template.version).to.be.not.undefined;
+          expect(template.version!.description).equals(VALID_VERSION.description);
+        });
+    });
+  });
+
+  let versionOneNumber: string;
+  let versionTwoNumber: string;
+  const versionOneDescription = `getTemplateAtVersion test v1 ${Date.now()}`;
+  const versionTwoDescription = `getTemplateAtVersion test v2 ${Date.now()}`;
+
+  describe('getTemplateAtVersion', () => {
+    before(async () => {
+      // obtain the current active template
+      let activeTemplate = await admin.remoteConfig().getTemplate();
+
+      // publish a new template to create a new version number
+      activeTemplate.version = { description: versionOneDescription };
+      activeTemplate = await admin.remoteConfig().publishTemplate(activeTemplate)
+      expect(activeTemplate.version).to.be.not.undefined;
+      versionOneNumber = activeTemplate.version!.versionNumber!;
+
+      // publish another template to create a second version number
+      activeTemplate.version = { description: versionTwoDescription };
+      activeTemplate = await admin.remoteConfig().publishTemplate(activeTemplate)
+      expect(activeTemplate.version).to.be.not.undefined;
+      versionTwoNumber = activeTemplate.version!.versionNumber!;
+    });
+
+    it('verfy that getTemplateAtVersion() returns the requested template version v1', () => {
+      return admin.remoteConfig().getTemplateAtVersion(versionOneNumber)
+        .then((template) => {
+          expect(template.etag).matches(/^etag-[0-9]*-[0-9]*$/);
+          expect(template.version).to.be.not.undefined;
+          expect(template.version!.versionNumber).equals(versionOneNumber);
+          expect(template.version!.description).equals(versionOneDescription);
+        });
+    });
+
+    it('verfy that getTemplateAtVersion() returns the requested template version v2', () => {
```

Comment:
Seems redundant.

Reply:
This was to confirm that we are not getting the most recent template and having two versions confirms it. I see your point though :)
Will remove the `expect`s here, but leave the templates for `listVersions` tests.